app/controllers/payments.js

```diff
@@ -1,19 +1,63 @@
 const config = require('config')
 const stripe = require('stripe')(config.stripe.api_secret)
+const plan = config.stripe.enthusiast_plan_id
+const User = require('../models/user.js')
+const logger = require('../../lib/logger.js')()
 
-exports.post = (req, res) => {
-  const amount = 1000
+exports.post = async (req, res) => {
+  let userId
+  let sub
+  let customer
+  try {
+    userId = req.body.userId
+    logger.info(`submitting payment for ${userId}`)
+    const { token: { email, id } } = req.body
 
-  stripe.customers.create({
-    email: req.body.stripeEmail,
-    source: req.body.stripeToken
-  })
-    .then(customer => console.log(customer) ||
-      stripe.charges.create({
-        amount,
-        description: 'Sample Charge',
-        currency: 'usd',
-        customer: customer.id
-      }))
-    .then(charge => console.log(charge) || res.render('main'))
+    customer = await stripe.customers.create({
+      email,
+      source: id,
+      description: `Subscriber for ${userId}`
+    })
+
+    sub = await stripe.subscriptions.create({
+      customer: customer.id,
+      items: [
+        {
+          plan
+        }
+      ]
+    })
+  } catch (err) {
+    logger.error(err)
+    res.status(500).send('Unexpected error while submitting payment.')
```

Comment:
See #1454, in regards to how we respond with status errors. In short, moving forward, all API responses should be JSON instead of strings.